libkbfs/folder_branch_ops.go

```diff
@@ -1724,7 +1729,8 @@ func isRecoverableBlockErrorForRemoval(err error) bool {
 
 func isRetriableError(err error, retries int) bool {
 	_, isExclOnUnmergedError := err.(ExclOnUnmergedError)
-	recoverable := isExclOnUnmergedError || isRecoverableBlockError(err)
+	recoverable := isExclOnUnmergedError || isRevisionConflict(err) ||
```

Comment:
I guess my main concern is that it's a bit confusing to have to remember that revision conflicts here (and a few other places) mean unmerged self-conflicts. There's also the possibility of bugs causing other types of conflicts to bubble up.

Can you make another error type, say, `UnmergedSelfConflictError`, and check for that here? And then in `finalizedMDWriteLocked` translate from a revision conflict from `PutUnmerged` into that error (and replace the TODO with a comment saying that `UnmergedSelfConflictError` is handled in `doMDWriteWithRetry`.